AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -140,18 +139,21 @@ + (MSTokenResult *)retrieveCachedToken:(NSString *)partitionName {
     NSString *tokenString = [MSKeychainUtil stringForKey:[MSTokenExchange tokenKeyNameForPartition:partitionName]];
     if (tokenString) {
       MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithString:tokenString];
+      if (![kMSTokenResultSucceed isEqualToString:tokenResult.status]) {
```

Comment:
Can we also validate the actual status code from the HTTP call that I assume is done earlier.

I caught this bug when the token endpoint was returning us a 400 code and we were still trying to parse the non-existent payload.

Reply:
In `performDbTokenAsyncOperationWithHttpClient` (MSTokenExchange) we should also validate the status code before proceeding in parsing the response.

Reply:
And in this method actually, we only read a token from the cache. If you look in the method I linked above, only valid tokens (status = succeeded) are cached so this test is not needed.